lib/browser/tag/each.js

```diff
@@ -170,79 +132,76 @@ export default function _each(dom, parent, expr) {
           item
         }, dom.innerHTML)
 
+        // link the original item to it's tag
+        if (isObject(_item))
+          defineProperty(_item, '_riot_id', tag._riot_id)
+
+        tagsMounted[tag._riot_id] = tag
+
         tag.mount()
-        domToInsert = tag.root
-        // this tag must be appended
-        if (i == tags.length) {
-          if (isVirtual)
-            makeVirtual(tag, frag)
-          else frag.appendChild(domToInsert)
-        }
-        // this tag must be insert
-        else {
-          if (isVirtual)
-            makeVirtual(tag, root, tags[i])
-          else root.insertBefore(domToInsert, tags[i].root)
-          oldItems.splice(i, 0, item)
-        }
 
-        tags.splice(i, 0, tag)
-        if (child) arrayishAdd(parent.tags, tagName, tag, true)
-        pos = i // handled here so no move
-      } else tag.update(item)
-
-      // reorder the tag if it's not located in its previous position
-      if (pos !== i && _mustReorder) {
-        // update the DOM
-        if (isVirtual)
-          moveVirtual(tag, root, tags[i], dom.childNodes.length)
```

Comment:
Ah, I didn't look it through, sorry. Now I get it. This is odd...
I was looking through the tests and there were some that relied on dom nodes being properly moved. Will try to fix it.